### ISSUE TITLE
RCv3 errors should raise `RequestError`

### ISSUE THREAD

**Issue by manishtomar** [rackerlabs/otter#826]
When RCv3 service errors, it raises `APIError`. It should wrap this in `RequestError` and raise that just like all other services do. This helps in recognizing which service has caused the error and is needed to run dashbards that depend on this.

**Comment by lvh** [comment]
Okay, so there's some discussion about this ticket. Do you agree the real resolution is "ServiceRequest raises UpstreamError", @manishtomar? Also, what, if anything, needs to be done about worker; and, how far can we delay this?

**Comment by manishtomar** [comment]
> Do you agree the real resolution is "ServiceRequest raises UpstreamError", @manishtomar?

Yes. Although I would like @cyli opinion on this as we should have one class used to report error in worker. That should be either `UpstreamError` or `RequestError`. For convergence we should use a better `UpstreamError`.  

> Also, what, if anything, needs to be done about worker; and, how far can we delay this?

I am sorry I didn't understand the question. I think we don't need to do anything more in it after this change (if that is what you are asking)

**Comment by cyli** [comment]
@manishtomar Sure, `UpstreamError` sounds good :)
